### ISSUE TITLE
Make remote reports show the user who sent the reports

### ISSUE THREAD

**Issue by Laurelai** [tootsuite/mastodon#7608]
Right now it doesnt and since remote reporting has been implemented its at least on my instance been used for nothing but frivolous reports, using the anonymity as a means of harassing me. Knowing who is sending them will allow me to suspend those users from my instance without having to suspend their whole instance 

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by nightpool** [comment]
Deanonymizing the remote user would be a huge privacy concern. I think requiring the remote mods to "forward" the original report (instead of users being able to do it without oversight) would be a good compromise here. (this was how the feature was originally proposed)

**Comment by Laurelai** [comment]
Its not a major privacy concern since any user on a remote instance can, without their admins knowledge or approval send remote anonymous harassment to any instance admin/mod team. The reality is that this privacy is being abused to cause harm. The privacy concerns have to give way to the safety of others in this case.

 I dont deserve anonymous harassment just so some hypothetical person can feel marginally safer. users on my instance have to send me reports locally with their usernames on them, if its not a privacy concern for them its not a concern for remote users.

Whoever decided that being able to send instance admins anonymous comments was a good idea didnt think this through.

**Comment by Hasimir** [comment]
Hang on, how come an instance admin can't set a policy of their own terms under which a report will be accepted?

What's wrong with an admin being able to set a policy to only accept identified reports from remote instances and if the reporter is notified of this before clicking submit then they can either choose to click the button or speak to their own instance's admin to forward it on their behalf?

**Comment by 0xa** [comment]
Reports can't be sent only remotely, so if someone harasses you through reports their admins will have seen it. (or they're ignoring reports, and at this point you may as well want to block the whole instance)
Keeping that in mind, I think anonymizing remote reports is an extremely important feature that could protect naive users from potentially hostile instance admins.

I think it would be best to let mods/admins decide to forward an anonymized report or not, as they surely would be in the best position to decide whether or not it makes sense depending on their relations in the fediverse.

**Comment by Laurelai** [comment]
Instance admins should have the ability at least to disable receiving remote reports yes? Because as it stands right now someone can hypothetically make a bunch of accounts on many instances and flood someone with frivolous reports. Someone could even automate it with a bot. Someone eventually will.

**Comment by 0xa** [comment]
Yes, I'd like to see that as an option along with silence/suspension of an instance, like the one to reject media but for reports. (and suspension should probably also block remote reports, i'm not sure of the current behavior since it was added later)

Having each instance's staff make a first validation of reports before forwarding would also greatly help by filtering out the spam at the source, so I'd much rather we do that _instead of_ exposing the original reporter's identity.

**Comment by Laurelai** [comment]
Or at the very least an option to click called "reject future reports from this user" and i wont have to see who they are.

**Comment by ThisIsMissEm** [comment]
>  Is there an easy way to DM another admin from the report UI, like "reply to this admin by DM"?

I think this would be a really good idea; Perhaps we can do federation of the Reports / Report Notes? That way the discussion happens inline, and actions that are taken on either instance are federated, so we can see "Target Instance silenced the user" on the source instance, for example.

**Comment by ThisIsMissEm** [comment]
I also agree with the idea of the admin/moderator approved forwarding of reports, rather than just automatically forwarding them.

**Comment by fsnk** [comment]
I remember @nightpool's comments in the discussion about federating anonymous abuse reports; they totally predicted how the system would be abused. 

As I see it, the main problem with anonymized reports on the recipient end is that it leaves me as an admin with no middle ground tools between just suffering abuse report spam or suspending an entire instance. 

Part of the problem is it seems like some admins may be loathe to make judgment calls on whether their users are abusing the anonymous nature of the report system to take malicious action against other instances. There may also be times when an admin is unavailable to deal with the issue. Many instances are small, and run as hobbies, and humans have lives. They have families, and vacations, they get sick, or they have pressing deadlines at work -- and a lot of abuse reports spam can be generated in a short period of time. (I expect we'll see bots doing this shortly, now that it's been exposed as a viable attack vector.) 

Obviously I think inter-admin communication and cooperation is the best option but the ability to just suspend remote reports on a per instance basis seems like a good compromise when remote admins are, for whatever reason, unwilling or unable to handle the attack from their end.

But just the ability to suspend reporting from a given instance doesn't fix the problem, because the decentralized nature of the fediverse means a handful of bad actors can create a few accounts on multiple - even many instances, and use them to send a low volume of spam reports from each instance which can fly under the radar of the local admin but become a flood on the receiving server. Add to that they can report multiple users on the remote instance, further camouflaging their reports on the local servers as legitimate. And if you can automate torrent spam, you can automate this kind of attack. 

This means an admin could have to potentially contact hundreds of other admins individually with the specific details and then wait while those admins research on their end and decide how to deal with the accounts causing the problem. All while just dealing with an influx from dozens of servers, or else suspending them. Right now, with suspending federation being the only self-defense, this would be a really good way to force another server to self-isolate. 

Of course *another* problem is currently the admin interface is inadequate for handling anything in batch, which means every single report has to be manually cleared. (This was evident in the recent spam account problems also.)

**Comment by Laurelai** [comment]
I told people that report harassment was a thing and to do something about it. Now it happened because of a celebrity. People now know they can report harass admins. Please fix this.

**Comment by Gargron** [comment]
The solution still isn't "Make remote reports show the user who sent the reports"

Because if you unknowingly report someone to evilnazi.town without realizing, you'll become their target. In fact, in the situation you are referring to, people were blaming Wil for reporting them, but they had no way of knowing if it was Wil. It didn't stop them from assuming, but the situation couldn't have been better if they'd seen who it really was. The people who thought that those people were crossing a line would become targets also.

2.5.0 includes a quick fix for a start: Duplicate reports (i.e. if another unresolved report about the same user exists) no longer generate e-mail notifications, and all staff users get a preference to disable e-mail notifications for reports altogether (finally)

From this point forward (2.6 roadmap) we will add report source blacklisting (i.e. no more reports from a certain local user, or certain remote domain), improve reports overview UI to add more batch operations and display a per-account summary instead of having 40 reports about the same user scattered all over the page. Also, adding a moderation API

**Comment by nightpool** [comment]
I still think it's really important to make remote report federation subject to local admin approval

**Comment by fsnk** [comment]
I think @nightpool has a point. Maybe that could be a per server option also, instead of a binary on/off setup we could use whitelisting/greylisting/blacklisting of reports, depending on the source server.

Like: accept all reports from server example.com / accept admin forwarded reports from server example.com / reject all reports from example.com

**Comment by fsnk** [comment]
A note about the suggestion to add "DM" to reports: our abuse handling policy is that we do not use Mastodon for this kind of conversation _at all_. Besides the privacy concerns and the possibility of accidentally blasting the discussion publicly across the network if someone fat fingers the visibility setting, the messaging setup isn't suited to keeping important conversations together and in a prominent place where they're easily referenced and searched.

**Comment by RetroAsgardian** [comment]
Here's a possible alternative solution:
What if reports had a federated discussion thread between the user who made the report, the local admins, and the remote admins? It'd help increase admin transparency, and allow admins to easily contact the report author to get additional information, while keeping it in an organized and separate place.

The report author's username doesn't have to be sent to the remote instance with this method, since it could just reference the report ID and the source instance would know which user it's talking about.

**Comment by tnonate** [comment]
> Or at the very least an option to click called "reject future reports from this user" and i wont have to see who they are.

I would really like to see an option like this. Today I got a report from someone who reported a comment that was CLEARLY not controversial or in violation of any kind of reasonable CoC - except maybe some strong language. This is clearly not a person I want wasting my time in the future crying about everyone out there who doesn't 1:1 match their opinions or whatever this person's complaint was. It also makes me wonder what happens in the future if some troll decides to keep spamming reports at me? Do I just have to defederate from the whole instance?

**Comment by ThisIsMissEm** [comment]
> What if reports had a federated discussion thread between the user who made the report, the local admins, and the remote admins?

Just seeing this, but it's exactly what I'm proposing with https://github.com/mastodon/mastodon/issues/27562 (fwiw, I added the original moderation notes functionality)

As for reports, I suspect having some sort of Reject on a Flag activity may make sense (the federated server declined to accept the report). There's probably still some way we can let the origin know "hey, we're not interested in reports like these"

**Comment by ThisIsMissEm** [comment]
Have taken the idea I proposed back in 2018, and opened it as a discussion on SocialHub: https://socialhub.activitypub.rocks/t/ability-to-distinguish-individual-users-behind-flag-activities-in-a-privacy-preserving-way/4235
